code/datums/components/combustible_flooder.dm

```diff
@@ -0,0 +1,56 @@
+/datum/component/combustible_flooder
+	// Gas type, molar count, and temperature. All self explanatory.
+	var/gas_name
+	var/gas_amount
+	var/temp_amount
+
+/datum/component/combustible_flooder/Initialize(gas_name, gas_amount, temp_amount)
+
+	src.gas_name = gas_name
+	src.gas_amount = gas_amount
+	src.temp_amount = temp_amount
+
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/attackby_react)
+	RegisterSignal(parent, COMSIG_ATOM_FIRE_ACT, .proc/flame_react)
+
+/datum/component/combustible_flooder/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_PARENT_ATTACKBY)
+	UnregisterSignal(parent, COMSIG_ATOM_FIRE_ACT)
+
+/// Do the flooding. overriden_temperature is the temperature we are overriding the initialized variable with.
+/datum/component/combustible_flooder/proc/flood(mob/user, overriden_temperature = temp_amount)
+	var/turf/open/flooded_turf = get_turf(parent)
+	flooded_turf.atmos_spawn_air("[gas_name]=[gas_amount];TEMP=[temp_amount]")
```

Comment:
Wait you don't use `overriden_temperature`.

Oh also I'd prefer setting `overriden_temperature` to `temp_amount` if nothing is passed in be done manually, since it's kinda unintuitive behavior

Reply:
Oh dear, the overriden_temperature there was a leftover from the plasma behaviour in the prev commit, removed.